web-ui/src/App.jsx

```diff
@@ -10,66 +10,79 @@ import { AppContextProvider } from './context/AppContext';
 import SnackBarWithContext from './components/snackbar/SnackBarWithContext';
 import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
 import { LocalizationProvider } from '@mui/x-date-pickers';
-import { createTheme, ThemeProvider } from '@mui/material/styles';
+
+import {
+  experimental_extendTheme as extendTheme,
+  Experimental_CssVarsProvider as CssVarsProvider
+} from '@mui/material/styles';
 
 import './App.css';
 
-const customHistory = createBrowserHistory();
+function getUserColorScheme() {
+  console.log('getting your scheme');
+  if (
+    window.matchMedia &&
+    window.matchMedia('(prefers-color-scheme: dark)').matches
+  ) {
+    return 'dark';
+  } else if (
+    window.matchMedia &&
+    window.matchMedia('(prefers-color-scheme: light)').matches
+  ) {
+    return 'light';
+  } else {
+    return 'light';
+  }
+}
 
-const theme = createTheme({
-  palette: {
-    primary: {
-      light: '#6085d9',
-      main: '#2559a7',
-      dark: '#003177',
-      contrastText: '#fff'
-    },
-    secondary: {
-      light: '#ffe8a2',
-      main: '#feb672',
-      dark: '#c88645',
-      contrastText: '#000'
+const theme = extendTheme({
+  cssVarPrefix: 'checkins',
+  colorSchemes: {
+    light: {
```

Comment:
Do light and dark ever contain properties other than palette?
I'm wondering if that layer of the hierarchy can be removed.

Reply:
In some cases they are scoped to a component, but for mui to respect the variables then palette must be used, I would have preferred to have had this without palette too but sometimes that's the nature of a design system.